ddtrace/_trace/processor/__init__.py

```diff
@@ -160,6 +162,10 @@ def process_trace(self, trace):
 
             if self.apm_opt_out:
                 chunk_root.set_metric(MK_APM_ENABLED, 0)
+                if APPSEC.PROPAGATION_HEADER not in chunk_root.get_tags():
```

Comment:
Would you want to move this code chunk below, after we sample the trace, right before we return it? Otherwise we may (outgoing request sent earlier causes sampling to run) more may not have a sampling_priority on the span. It seems like it could vary what `root_ctx._previous_sampling_priority` returns.